src/column/reader.rs

```diff
@@ -511,6 +539,71 @@ mod tests {
   test!(test_read_dict_v2_int64, i64, dict_v2, MAX_DEF_LEVEL, MAX_REP_LEVEL,
     NUM_PAGES, NUM_LEVELS, 16, 0, 3);
 
+  // test read_batch functionality, how it handles different input parameters
+  fn test_read_batch_int32(
```

Comment:
nit: could you move this down after all the test cases - makes it easier to read :)